lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -189,11 +189,8 @@ private function isAutoIncrementSequenceInSchema($schema, $sequence)
      */
     public function diffSequence(Sequence $sequence1, Sequence $sequence2)
     {
-        if ($sequence1->getAllocationSize() != $sequence2->getAllocationSize()) {
-            return true;
-        }
-
-        return $sequence1->getInitialValue() !== $sequence2->getInitialValue();
+        return intval($sequence1->getAllocationSize()) !== intval($sequence2->getAllocationSize())
+            || intval($sequence1->getInitialValue()) !== intval($sequence2->getInitialValue());
```

Comment:
The `Sequence` is a DBAL-level type which should enforce proper types of its internal values. If it doesn't, it should be fiexed in the `Sequence`, not in `Comparator`. Otherwise, we'll end up having type casting across the entire codebase.